common/membership/ringpop/service_resolver.go

```diff
@@ -234,6 +234,20 @@ func (r *serviceResolver) MemberCount() int {
 	return len(hosts)
 }
 
+func (r *serviceResolver) AvailableMemberCount() int {
+	_, hosts := r.ring()
+	n := 0
+	for _, host := range hosts {
+		if drainingStr, ok := host.Label(drainingKey); ok {
+			if draining, _ := strconv.ParseBool(drainingStr); draining {
+				continue
+			}
+		}
```

Comment:
If we think this is too slow to call all the time, we can precompute the result when the ring changes and use that here

Reply:
Probably not a concern. Why not just do `return len(r.AvailableMembers())` ?

Reply:
Doing a bunch of lookups may be fine but I think it'd be nice to avoid allocation